packages/compiler-core/src/transforms/vOn.ts

```diff
@@ -50,7 +50,10 @@ export const transformOn: DirectiveTransform = (
       const eventString =
         node.tagType === ElementTypes.COMPONENT ||
         rawName.startsWith('vnode') ||
-        !/[A-Z]/.test(rawName)
+        (!/[A-Z]/.test(rawName) ||
+          // See issue #6900,Ensure consistent behavior of
+          // @update:modelValue and @update:model-value
+        rawName === 'update:modelValue')
```

Comment:
the problem isn't specific to `update:modelValue`, it happens with any `v-on` that has an upperCase letter in it

Reply:
I've noticed that the result is correct when the node type is `component` , and for this `bug` perhaps we can compile the same result when the node type is `slot` as if the node type were `component` . When the node type is `element` , the original logic is preserved, meaning that the presence of uppercase characters resolves to custom events for custom elements, as described in the comments on lines 59-60.